jupyterlab_gitsync/src/service/request_api.ts

```diff
@@ -0,0 +1,31 @@
+import { ServerConnection } from '@jupyterlab/services';
+import { URLExt } from '@jupyterlab/coreutils';
+
+/**
+ * Call the extension APIs
+ *
+ * @param extension: Handler extension for url
+ * @param init: Initial values for the request
+ * @returns: The response body interpreted as JSON
+ */
+
+export async function requestAPI(extension = '', init: RequestInit = {}) {
+  const settings = ServerConnection.makeSettings();
+  const url = URLExt.join(settings.baseUrl, 'jupyterlab_gitsync/', extension);
+
+  let response: Response;
+
+  try {
+    response = await ServerConnection.makeRequest(url, init, settings);
+  } catch (error) {
+    throw new ServerConnection.NetworkError(error);
+  }
+
+  const data = await response.json();
+
+  if (!response.ok) {
```

Comment:
should check this before collecting data. Don't know about ts HTTP client, some libraries may fail to collect JSON data if there's invalid response.

Reply:
fixed so data is gotten after checking response.ok.